### PR TITLE
[FEATURE][SC-40667] Add range modifiers for year calendar

### PR DESCRIPTION
**Pull request by KevinFabre-ods**
- Add missing range modifiers (`--selecting-range-start`, `--selecting-range-end`) for month calendar
- Add range modifiers for year calendar
   - Which made it possible to expose `onYearMouseEnter` `onYearMouseLeave` as props
   
   
   
   

https://github.com/opendatasoft/react-datepicker/assets/117300300/00587d71-6e4b-4d82-86f5-803904eef809

 Internal: [SC-40667](https://app.shortcut.com/opendatasoft/story/40667/)
 
 related issues:
 - https://github.com/Hacker0x01/react-datepicker/issues/4048
 - https://github.com/Hacker0x01/react-datepicker/issues/2719

### CODE REVIEW

File: src/year.jsx [hunk 1]
```diff
@@ -65,6 +74,85 @@ export default class Year extends React.Component {
...
+    const _year = utils.setYear(newDate(), y);
+    const selectingDate = this.props.selectingDate ?? this.props.preSelection;
+
+    if (selectsEnd || selectsRange) {
+      return utils.isSameYear(_year, selectingDate);
+    } else {
+      return utils.isSameYear(_year, endDate);
+    }
+  };
+
```
Comment (wmai) [inline src/year.jsx]: The code suits me perfectly as it is, but if we wanted to reduce a little bit of the boilerplate code and maybe gain some readability we could create two helper methods `getSelectingDate()` and `getYear()`:

```JS
getSelectingDate = () => this.props.selectingDate ?? this.props.preSelection;

getYear = (y) => utils.setYear(newDate(), y);

// Pretty one liners ^^

isRangeStart = (y) => this.props.startDate && this.props.endDate && utils.isSameYear(this.getYear(y), this.props.startDate);

isRangeEnd = (y) => this.props.startDate && this.props.endDate && utils.isSameYear(this.getYear(y), this.props.endDate);

isInRange = (y) => utils.isYearInRange(y, this.props.startDate, this.props.endDate);

// ...
```

**Review by wmai — APPROVED** [review]
Great work 🙏 Looks perfectly good to me! I've only left a very minor comment you can ignore if you prefer.

File: src/year.jsx (hunk 1 above)
Comment (KevinFabre-ods) [reply src/year.jsx]: will do 👍

**Comment by KevinFabre-ods** [comment]
Closing it as we are opening a PR in main repo.

**Pull request closed without merging**